ci/scripts/glib_test.sh

```diff
@@ -63,7 +65,14 @@ test_subproject() {
            env DYLD_LIBRARY_PATH="${DYLD_LIBRARY_PATH}" \
            ruby test/run.rb
     bundle exec rake build
-    gem install --install-dir "${build_dir}/gems" pkg/*.gem
+
+    echo "Testing Gem"
+    local gem_flags=""
+    if [[ "$(uname)" = "Darwin" ]]; then
+        gem_flags='-- --with-cflags="-D_LIBCPP_DISABLE_AVAILABILITY" --with-cppflags="-D_LIBCPP_DISABLE_AVAILABILITY"'
+    fi
+
+    gem install --install-dir "${build_dir}/gems" pkg/*.gem -- ${gem_flags}
```

Comment:
```suggestion
    gem install --install-dir "${build_dir}/gems" pkg/*.gem ${gem_flags}
```

Reply:
I had problems with the dependency-check Rakefile getting the arguments and then erroring without the double-layer of  `--` - is there a better way to pass this?

Reply:
See this CI run: https://github.com/apache/arrow-adbc/actions/runs/3801840679/jobs/6466736182#step:20:107

from this commit: https://github.com/apache/arrow-adbc/pull/305/commits/257e12b60ad3e9c3ec00d33895dc4d94fa3f9508

Reply:
Ah, OK. I understand.
How about `export MACOSX_DEPLOYMENT_TARGET=10.14` instead of `--with-cflags...`?

Reply:
```bash
export MACOSX_DEPLOYMENT_TARGET=10.14
gem install --install-dir "${build_dir}/gems" pkg/*.gem
```

Reply:
ah, that would be much better!

Reply:
Updated.

Reply:
Sorry... `MACOSX_DEPLOYMENT_TARGET` didn't work...
I'll look into this later. We can use `-- -- --with-cflags...` for now.

Reply:
It seems that Red Arrow needs to specify `-mmacosx-version-min=${MACOSX_DEPLOYMENT_TARGET}` explicitly. I'll try it later. (This doesn't block this pull request.)